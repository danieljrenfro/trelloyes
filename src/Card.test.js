import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Card from './Card';


describe('Card Component', () => {

  it('renders without crashing with no props', () => {
    let div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without crashing with props', () => {
    let div = document.createElement('div');
    ReactDOM.render(<Card title='lorem' content='ipsum' />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the card to the UI as expected with no data', () => {
    const tree = renderer
      .create(<Card />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the card to the UI as expected with data', () => {
    const tree = renderer
      .create(<Card title='lorem' content='ipsum' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});