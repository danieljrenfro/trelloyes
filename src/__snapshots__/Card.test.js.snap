// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the card to the UI as expected with data 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    lorem
  </h3>
  <p>
    ipsum
  </p>
</div>
`;

exports[`Card Component renders the card to the UI as expected with no data 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
